### ISSUE TITLE
[BUG] WebGL with dashed line is not working when click_policy is muted

### ISSUE THREAD

**Issue by hoxbro** [bokeh/bokeh#13195]
### Software versions

``` yaml
Python version      :  3.11.3 | packaged by conda-forge | (main, Apr  6 2023, 08:57:19) [GCC 11.3.0]
IPython version     :  8.14.0
Tornado version     :  6.3.2
Bokeh version       :  3.1.1
BokehJS static path :  /home/shh/miniconda3/envs/holoviz/lib/python3.11/site-packages/bokeh/server/static
node.js version     :  v18.15.0
npm version         :  9.5.0
Operating system    :  Linux-6.2.6-76060206-generic-x86_64-with-glibc2.35
```

### Expected behavior

That it works like without dashed line.

### Observed behavior

Dashed line with `click_policity="mute"` in WebGL does not work. 

### Example code

```Python
import numpy as np
from bokeh.plotting import figure, show
from bokeh.io import output_notebook
output_notebook()

x = np.linspace(0, 4*np.pi, 100)
y = np.sin(x)

p = figure(output_backend="webgl")
p.line(x, y, legend_label="2 * sin(x)", line_dash=(4,4), color="orange")
p.legend.click_policy="mute"
show(p)
```

### Stack traceback or browser console output

``` console
Uncaught (in promise) TypeError: props.length_so_far is undefined
    a_length_so_far https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.1.1.js:275
    draw https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.1.1.js line 5488 > Function:563
    REGLCommand https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.1.1.js:8934
    _draw_impl https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.1.1.js:10690
    draw https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.1.1.js:11232
    render https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.1.1.js:10789
    render https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:22947
    _render https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:22537
    render https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:9936
    _paint_levels https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:43992
    _actual_paint https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:43964
    paint https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:43904
    repaint https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:43897
    throttled_paint https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:43443
    later https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:44097
    fn https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:44111
    fn https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:44089
    schedule_paint https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:43398
    request_paint https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:43384
    request_paint https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:9908
    request_render https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:9905
    render https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:22368
    new_slot https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:7723
    emit https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:2736
    emit https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:2745
    _setv https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:2506
    setv https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:2533
    set https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:2254
    fn https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:27459
    on_hit https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:27467
    __trigger https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:33074
    _trigger https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:32951
    _tap https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:33237
    _configure_hammerjs https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:32723
    emit https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:35476
    emit https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:35134
    tryEmit https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:34619
    _timer https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:35114
    boundFn https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:33445
    setTimeout handler*setTimeoutContext https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:33297
    process https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:35112
    recognize https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:34656
    recognize https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:35343
    inputHandler https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:33740
    PEhandler https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:34070
    domHandler https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.js:33663
bokeh-gl-3.1.1.js:275:21
```

### Screenshots

https://github.com/bokeh/bokeh/assets/19758978/65fb9048-1772-48d6-93c5-8c79cb32ea57

**Comment by hoxbro** [comment]
Could be fixed by: https://github.com/bokeh/bokeh/pull/13175

**Comment by ianthomas23** [comment]
Yes, I came across this when doing the multiline PR #13175, which does indeed *intend* to fix it. I will have to confirm that it actually does fix it!

**Comment by mattpap** [comment]
Let's make sure the PR fixing this adds tests for derived glyphs (selections, etc.)

**Comment by ianthomas23** [comment]
I have manually checked that this is indeed fixed by #13175. I'll set it back to WIP until I've added explicit tests for this.

**Comment by ianthomas23** [comment]
Closing as fixed by #13175 and that PR includes explicit tests for this use case.
